### PR TITLE
python310Packages.sphinx-book-theme: Mark as broken

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Package missing Sphinx 5 support and is therefore incompatible https://github.com/executablebooks/sphinx-book-theme/issues/592

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/sphinx-book-theme/default.nix [hunk 1]
```diff
@@ -23,6 +23,9 @@ buildPythonPackage rec {
     sha256 = "9685959dbbb492af005165ef1b9229fdd5d5431580ac181578beae3b4d012d91";
   };
 
+  postPatch = ''
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinx-book-theme/default.nix:28]: What is this?

File: pkgs/development/python-modules/sphinx-book-theme/default.nix [hunk 2]
```diff
@@ -36,5 +39,8 @@ buildPythonPackage rec {
     homepage = "https://github.com/executablebooks/sphinx-book-theme";
     license = licenses.bsd3;
     maintainers = with maintainers; [ marsam ];
+    # Missing sphinx 5.X support
+    # https://github.com/executablebooks/sphinx-book-theme/issues/592
+    broken = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinx-book-theme/default.nix:44]: If this wouldn't be a wheel I would suggest to patch the source instead.

**Comment by urkud** [comment]
Can we temporarily restore `sphinx-4.x`?

**Comment by SuperSandro2000** [comment]
> Can we temporarily restore `sphinx-4.x`?

Having multiple versions is problematic in python packages and we should avoid this as much as possible.

**Comment by drupol** [comment]
I can confirm that this package is not compatible with Sphinx 5.

File: pkgs/development/python-modules/sphinx-book-theme/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinx-book-theme/default.nix:28]:
```suggestion
```

**Comment by drupol** [comment]
The sphinx-book-theme has been updated, we will need to update this soon: https://github.com/executablebooks/sphinx-book-theme/issues/592

**Pull request merged**